src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/AbstractFlowScanner.java

```diff
@@ -307,6 +308,25 @@ public FlowNode findFirstMatch(@CheckForNull FlowExecution exec, @Nonnull Predic
         return nodes;
     }
 
+    /** Convenience method to get the list all flownodes in the iterator order. */
+    @Nonnull
+    public List<FlowNode> allNodes(@CheckForNull Collection<FlowNode> heads) {
+        if (!setup(heads)) {
+            return Collections.EMPTY_LIST;
```

Comment:
Always use factories like `emptyList()` rather than the constants, so you can avoid `@SuppressWarnings("unchecked")`.